src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -193,12 +192,17 @@ uint32_t CDC_Receive_DATA(uint8_t* recvBuf, uint32_t len)
 {
     uint32_t count = 0;
 
-    while (usbAvailable() && count < len) {
-        recvBuf[count] = usbData.buffer[usbData.bufferOutPosition];
-        usbData.bufferOutPosition = (usbData.bufferOutPosition + 1) % USB_RX_BUFSIZE;
+    while (APP_Tx_ptr_out != APP_Tx_ptr_in && count < len) {
+        recvBuf[count] = APP_Tx_Buffer[APP_Tx_ptr_out];
+        APP_Tx_ptr_out = (APP_Tx_ptr_out + 1) % USB_RX_BUFSIZE;
         count++;
         receiveLength--;
     }
+
+    if (!receiveLength) {
+        receiveLength = APP_Tx_ptr_out != APP_Tx_ptr_in;
```

Comment:
That situation almost certainly will never occur, but for the purposes of this fix there are much better ways of doing this - but it would require redoing serial_vcp_usb - and I don't want to touch F1 and F3 targets just yet (will do it separately).